src/components/People/PeopleTable.tsx

```diff
@@ -0,0 +1,249 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import type { Person } from '../../types';
+import { TitleTableHeaders } from '../../types';
+import { getPreparedPeople } from '../../utils/getPreparedPeople';
+import { getSearchWith } from '../../utils/searchHelper';
+import { Loader } from '../Loader';
+import { PeopleFilters } from './PeopleFilters';
+import { PersonItem } from './PersonItem';
+import { SearchLink } from '../SearchLink';
+
+const DESCENDING_ORDER = 'desc';
+
+export const PeopleTable = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isMatchSearch, setIsMatchSearch] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(person => {
+        setPeople(person);
+      })
+      .catch(() => {
+        setIsError(true);
+        setTimeout(() => setIsError(false), 3000);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || null;
+
+  const sex = searchParams.get('sex') || null;
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const showTable = !isLoading && !isError && isMatchSearch;
+  const showSideBar = !isLoading && !isError;
+
+  const setSearchWith = (params: any) => {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+  };
+
+  const handleQueryChange = (selectQuery: string) => {
+    setSearchWith({ query: selectQuery });
+  };
+
+  const handleSexChange = (selectSex: string | null) => {
+    setSearchWith({ sex: selectSex });
+  };
+
+  const toggleCenturies = (selectCentury: string) => {
+    return centuries.includes(selectCentury)
+      ? centuries.filter(century => century !== selectCentury)
+      : [...centuries, selectCentury];
+  };
+
+  const handleSortChange = (sortBy: string) => {
+    const isSelected = sortBy === sort;
+
+    if ((!sort && !order) || !isSelected) {
+      return { sort: sortBy, order: null };
+    }
+
+    if (sort && !order && isSelected) {
+      return { sort: sortBy, order: DESCENDING_ORDER };
+    }
+
+    if (sort && order) {
+      return { sort: null, order: null };
+    }
+
+    return { sort: null, order: null };
+  };
+
+  const preparedPeople = getPreparedPeople(people);
+
+  const getSortedPeople = () => {
```

Comment:
if it's quite a big reusable function it makes sense to move it out of components and accept all parameters you need as arguments. But in this case would be even better to use `useMemo`